api/app/models/project.rb

```diff
@@ -82,6 +82,7 @@ class Project < ApplicationRecord
 
   # Callbacks
   after_commit :trigger_creation_event, on: [:create]
+  before_update :set_last_updated_at
```

Comment:
I think I understand what you're doing here. You're trying to avoid updating the update timestamp for a project when the draft mode changes. This approach feels wrong to me. None of this update timestamp logic feels like a proper concern for the Project model.

Rails 5 lets you not update a timestamp when you save the model by passing a touch: false option to the save method. For example: `project.save(touch: false)`

I think you should drop all this timestamp logic and just handle this in the project controller's update method. Currently, you have this:

`::Updaters::Project.new(project_params).update(@project)`

Why not change the `update` method in `Updaters` so it accepts a second options argument. In the controller, if the draft mode is the only thing being changed, you could pass `touch: false` to the update method, which could in turn pass the touch false option to the save method. This seems simpler and more reusable to me.